src/MBrace.Azure/Management.fs

```diff
@@ -338,177 +479,331 @@ module private Details =
-                   do! info (sprintf "using MBrace version tag %s" mbraceVersion)
-                   let uri = defaultArg CloudServicePackage (urlForPackage mbraceVersion vmSize)
-                   use wc = new System.Net.WebClient() 
-                   let tmp = System.IO.Path.GetTempFileName() 
-                   do! info (sprintf "downloading cloud service package from %s" uri)
-                   wc.DownloadFile(uri, tmp)
-                   return tmp, mbraceVersion
-            }
-           let! client = Credentials.connectToAzure Subscription pubSettingsFile 
-           let clusterName = defaultArg ClusterName (generateResourceName())
-           do! info (sprintf "using cluster name %s" clusterName)
-           let! storageAccountName = 
-             trial {
-               match StorageAccount with 
-               | None -> return! Storage.getDefaultMBraceStorageAccountName region client
-               | Some s -> return s
-             }
-           let! storageConnectionString = client |> Storage.getMBraceStorageConnectionString storageAccountName 
-           do! info (sprintf "using storage account name %s" storageAccountName)
-           do! info (sprintf "using storage connection string %s..." storageConnectionString.[0..20])
-           let! serviceBusNamespace, serviceBusConnectionString = client |> ServiceBus.getDefaultMBraceNamespace region
-           do! info (sprintf "using service bus namespace %s" serviceBusNamespace)
-           do! info (sprintf "using service bus connection string %s..." serviceBusConnectionString.[0..20])
-
-           let numInstances = defaultArg VMCount 2
-           let config = Clusters.buildMBraceConfig clusterName numInstances storageConnectionString serviceBusConnectionString
-    
-
-           let clusterLabel = defaultArg ClusterLabel (sprintf "MBrace cluster %s, package %s"  clusterName infoString)
-           do! Clusters.createMBraceCluster(clusterName, clusterLabel, region, packagePath, config, storageAccountName, storageConnectionString, serviceBusNamespace, serviceBusConnectionString) client
-
-           let config = Configuration(storageConnectionString, serviceBusConnectionString)
-
-           return config
-        }  |> Trial.returnOrFail
-
-    static member DeleteCluster(pubSettingsFile, clusterName, ?Subscription) =
-       trial { 
-           let! client = Credentials.connectToAzure Subscription pubSettingsFile 
-           return! Clusters.deleteMBraceCluster clusterName client
-
-        }  |> Trial.returnOrFail
-
-    static member GetClusters(pubSettingsFile, ?Subscription) =
-       trial { 
-           let! client = Credentials.connectToAzure Subscription pubSettingsFile 
-           let response = Clusters.listRunningMBraceClusters client
-           return response
-
-        }  |> Trial.returnOrFail
+        let deploy (slot : DeploymentSlot) (deployParams : DeploymentCreateParameters) (client : SubscriptionClient) = async {
```

Comment:
No, I've removed it from the API since the CreateAsync method is not working for me.